Formula/n/ncmdump.rb

```diff
@@ -0,0 +1,26 @@
+class Ncmdump < Formula
+  desc "Convert Netease Cloud Music ncm files to mp3/flac files"
+  homepage "https://github.com/taurusxin/ncmdump"
+  url "https://github.com/taurusxin/ncmdump/archive/refs/tags/1.2.1.tar.gz"
+  sha256 "a1bd97fd1b46f9ba4ffaac0cf6cf1e920b49bf6ec753870ad0e6e07a72c2de2d"
+  license "MIT"
+
+  depends_on "taglib"
+
+  def install
+    if OS.mac?
+      if Hardware::CPU.arm?
+        system "make", "macos-arm64"
+      elsif Hardware::CPU.intel?
+        system "make", "macos-intel"
+      end
+    elsif OS.linux?
+      system "make", "linux"
+    end
+    bin.install "ncmdump"
+  end
+
+  test do
+    system "#{bin}/ncmdump", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
What about if I download a test file from GitHub repository and run the test?

Reply:
yeah, that works, you can find many examples in this repo